megalinter/llm_advisor.py

```diff
@@ -0,0 +1,209 @@
+    def SUPPORTED_PROVIDERS(self) -> Dict[str, str]:
+        """Backwards compatibility property for supported providers"""
+        return self.get_supported_providers()
+
+    def get_fix_suggestions(
+        self, linter_name: str, linter_output: str
+    ) -> Dict[str, Any]:
+        """
+        Get AI-powered fix suggestions for linter errors
+
+        Args:
+            linter_name: Name of the linter
+            linter_output: Full raw output from the linter
+
+        Returns:
+            Dictionary containing a single fix suggestion and metadata
+        """
+        if not self.is_available():
+            return {"enabled": False, "suggestion": None}
+
+        return self._get_suggestion_from_raw_output(linter_name, linter_output)
+
+    def _get_suggestion_from_raw_output(
+        self, linter_name: str, linter_output: str
+    ) -> Dict[str, Any]:
+        """
+        Get AI suggestion directly from raw linter output
+
+        Args:
+            linter_name: Name of the linter
+            linter_output: Full raw output from the linter
+
+        Returns:
+            Dictionary containing a single fix suggestion and metadata
+        """
+        try:
+            # Build a prompt for analyzing the raw output
+            prompt = self._build_raw_output_prompt(linter_name, linter_output)
+            system_prompt = self._get_system_prompt()
+
+            # Get response from LLM provider
+            suggestion_text = self.provider.invoke(prompt, system_prompt)
+
+            # Return as a single suggestion for the raw output
+            suggestion = {"linter": linter_name, "suggestion": suggestion_text.strip()}
+
+            return {
+                "enabled": True,
+                "provider": self.provider_name,
+                "model": self.model_name,
+                "suggestion": suggestion,
+            }
+
+        except Exception as e:
+            logging.warning(f"Failed to get suggestion from raw output: {str(e)}")
+            return {
+                "enabled": True,
+                "provider": self.provider_name,
+                "model": self.model_name,
+                "suggestion": None,
+            }
+
+    def _get_system_prompt(self) -> str:
+        """Get the system prompt for the LLM"""
+        return """You are an expert code reviewer and linter error analyst. Your job is to help developers understand and fix linting errors in their code.
+
+For each linter error, provide:
+1. A clear explanation of what the error means
+2. Why this error occurs
+3. Specific, actionable steps to fix it
+4. If applicable, a code example showing the fix
+5. Best practices to prevent similar errors
+
+Keep your responses concise but comprehensive. Focus on practical solutions that developers can immediately apply.
+
+Your response must not exceed 1000 characters, so prioritize the most critical issues and solutions. If there are multiple errors, focus on the most common or critical ones first."""
```

Comment:
The 1000 characters seemed to be a configurable parameter, but is hardcoded here

Reply:
@echoix with the limits on github PR comments, i'm not sure it's safe to allow to override it